Contents/Code/__init__.py

```diff
@@ -37,24 +53,19 @@ def search_na(results, media_title, year, lang):
     searchURL = EXC_BASEURL + 'dev/' + query_actor
     Log('Search URL: ' + searchURL)
     search_results = HTML.ElementFromURL(searchURL)
-  
-  xp = '''//select//Option[text()[contains(translate(., "%s", "%s"), "%s")]]//@value''' % (na_url_site.upper(), na_url_site.lower(), na_url_site.lower())
-  Log('xPath: ' + xp)
+
   try:
+    searchURL = find_option_value(searchURL, search_results, search)
+  except:
     try:
-      searchURL = search_results.xpath(xp)[0]
+      searchURL = find_option_value(searchURL, search_results, re.sub(r'[\'\"]', '', search))
     except:
-      xp = '''//select//option[text()[contains(translate(., "%s", "%s"), "%s")]]//@value''' % (na_url_site.upper(), na_url_site.lower(), na_url_site.lower())
+      search_results = HTML.ElementFromURL(searchURL + '/sites/')
+      xp = xpath_prepare('//a[text()[contains(translate(., "$u", "$l"), "$s")]]//@href', search)
       Log('xPath: ' + xp)
       searchURL = search_results.xpath(xp)[0]
```

Comment:
so you're totally right, in my attempt to change/fix this I decided not to take care of that as it's already been that way for sometime. 

I'd like to refactor some of this stuff out, but for now just taking baby steps